syntax = "proto3";

package envoy.extensions.bootstrap.reverse_tunnel;

// Internal proto definitions for reverse connection handshake protocol.
// These messages are used internally by the reverse tunnel extension
// and are not exposed to users.

// Config sent by the local cluster as part of the Initiation workflow.
// This message combined with message 'ReverseConnHandshakeRet' which is
// sent as a response can be used to transfer/negotiate parameter between the
// two envoys.
message ReverseConnHandshakeArg {
  // Tenant UUID of the local cluster.
  string tenant_uuid = 1;

  // Cluster UUID of the local cluster.
  string cluster_uuid = 2;

  // Node UUID of the local cluster.
  string node_uuid = 3;
}

// Config used by the remote cluster in response to the above 'ReverseConnHandshakeArg'.
message ReverseConnHandshakeRet {

  enum ConnectionStatus {
    REJECTED = 0;
    ACCEPTED = 1;
  }

  // Tracks the status of the reverse connection initiation workflow.
  ConnectionStatus status = 1;

  // This field can be used to transmit success/warning/error messages
  // describing the status of the reverse connection, if needed.
  string status_message = 2;
}
